packages/admin-ui/src/app/settlements/settlements.matrices.component.ts

```diff
@@ -87,7 +87,13 @@ export class SettlementsMatricesComponent implements OnInit, OnDestroy {
 		).subscribe((matricesResult) => {
 			console.log("SettlementsMatricesComponent search - got MatricesSearchResult");
 
-			this.matrices.next(matricesResult.items || []);
+			if (!matricesResult || !matricesResult.items) {
```

Comment:
@ZweHMyat  
I think the problem here is the SettlementsService.searchMatrices() responding with undefined on subscriber.next(); when not found 404. 
It should return a proper MatrixSearchResults structure with empty items array.

Same for SettlementsService.getMatrices()

Reply:
Got it I will fix it in the service. Since I have to add pagination to the matrix page so I will combine this error fix with the pagination code and create a new PR again. I will close this one. Thank you.